src/librbd/LibrbdWriteback.cc

```diff
@@ -163,6 +168,8 @@ namespace librbd {
 				 object_no, off, objectx, object_overlap,
 				 bl, snapc, snap_id,
 				 req_comp);
+    if (m_ictx->closing)
+      req->set_op_flags(LIBRADOS_OP_FLAG_FADVISE_DONTNEED);
```

Comment:
@majianpeng Sounds good to me to remove this patch.  As for adding a way to disable the cache, am I correct in assume you want to disable it per-image regardless of where the image is used?  If so, there is a forthcoming change to RBD which adds flags to images -- disable cache could be a new flag.